src/debug.cc

```diff
@@ -71,19 +98,359 @@ class DebugFile
     FILE *file_ = nullptr; ///< opened "real" file or nil; never stderr
 };
 
-/// configured cache.log file or stderr
+/// meta-information of a Finish()ed debugs() message
+class DebugMessageHeader
+{
+public:
+    DebugMessageHeader(const DebugRecordCount aRecordNumber, const Debug::Context &);
+
+    DebugRecordCount recordNumber; ///< LogMessage() calls before this message
+    time_t timestamp; ///< approximate debugs() call time
+    int section; ///< debugs() section
+    int level; ///< debugs() level
+    bool forceAlert; ///< debugs() forceAlert flag
+};
+
+// Avoid SBuf for CompiledDebugMessageBody:
+// * SBuf's own debugging may create a lot of reentrant debugging noise.
+// * Debug::Context::buf is an std::string-based STL ostream. Converting its
+//   buf() result to a different kind of string may increase complexity/cost.
+// TODO: Consider switching to a simple fixed-size buffer and a matching stream!
+/// The processed "content" (i.e. the last parameter) part of a debugs() call.
+using CompiledDebugMessageBody = std::string;
+
+/// a fully processed debugs(), ready to be logged
+class CompiledDebugMessage
+{
+public:
+    using Header = DebugMessageHeader;
+    using Body = CompiledDebugMessageBody;
+
+    CompiledDebugMessage(const Header &aHeader, const Body &aBody);
```

Comment:
aHeader and aBody parameter names are redundant. please remove
```suggestion
    CompiledDebugMessage(const Header &, const Body &);
```

Reply:
Fixed in 28393b0.